File: Simulator-CSharp/Components/ScreenPad.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace ATM.Components
{
    public partial class ScreenPad : UserControl
    {
        public ScreenPad()
        {
            InitializeComponent();
        }

        private int actionPressed;
        public int ActionPressed
        {
            get { return actionPressed; }
        }

        public event EventHandler Pressed;
       
        private void buttonAction_Click(object sender, EventArgs e)
        {
            actionPressed = Convert.ToInt32(((Control)sender).Tag);
            if (Pressed != null)
                Pressed(this, EventArgs.Empty);
        }

    }
}

File: Simulator-VB/Components/ScreenPad.vb
Public Class ScreenPad

    Private _ActionPressed As Integer
    Public ReadOnly Property ActionPressed As Integer
        Get
            Return _ActionPressed
        End Get
    End Property

    Public Event Pressed As EventHandler

    Private Sub ButtonAction_Click(sender As Object, e As EventArgs) Handles ButtonAction4.Click, ButtonAction3.Click, ButtonAction2.Click, ButtonAction1.Click
        _ActionPressed = CInt(sender.Tag)
        RaiseEvent Pressed(Me, EventArgs.Empty)
    End Sub
End Class
